pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -252,35 +254,54 @@ cpdef inline object precision_from_unit(object unit):
         int64_t m
         int p
 
-    if unit == 'Y':
-        m = 1000000000L * 31556952
+    if unit is None:
+        m = 1L
+        p = 0
+        return m, p
+
+    unit, stride = _base_and_stride(unit)
+
+    # Normalize old or lowercase codes to standard offset aliases.
```

Comment:
I am not 100% sure we are actually testing all of these, can you see where we are and validate